kml/types.cpp

```diff
@@ -30,6 +30,56 @@ void SetBookmarksMinZoom(FileData & fileData, double countPerTile, int maxZoom)
   minZoomQuadtree.SetMinZoom(countPerTile, maxZoom, setMinZoom);
 }
 
+bool MultiGeometry::IsValid() const
+{
+  if (m_lines.empty())
+    return false;
+  if (!HasTimestamps())
+    return true;
+  for (size_t i = 0; i < m_lines.size(); ++i)
+  {
+    if (HasTimestampsFor(i) && m_lines[i].size() != m_timestamps[i].size())
```

Comment:
Where is `IsValid` method called? Decoding KML should not allow inconsistent timestamp vs points sizes. Meaning that this check should not be needed anywhere else, except when we validate the just loaded KML.

Reply:
This method is called multiple times in the different parts of the code.
You're wright! I'll bring back the old implementation.